### PR TITLE
Support for multiple embedding providers (Huggingface, etc.)

### PR DESCRIPTION
**Pull request by Abhinav-Naikawadi**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -64,10 +65,12 @@ anthropic = [
     "anthropic >= 0.2.6"
 ]
 huggingface = [
-    "transformers >= 4.25.0"
+    "transformers >= 4.25.0",
+    "accelerate == 0.20.3"
```
Comment (Abhinav-Naikawadi) [inline pyproject.toml:69]: This dependency was needed for hugging face pipelines support

File: pyproject.toml [hunk 2]
```diff
@@ -35,6 +35,7 @@ dependencies = [
     "torch >= 1.10.0",
     "matplotlib >= 3.5.0",
     "wget >= 3.2",
+    "ipywidgets == 8.0.6",
```
Comment (Abhinav-Naikawadi) [inline pyproject.toml:38]: This dependency was needed for jupyter notebook support for sentence transformers progress bars

File: src/autolabel/configs/config.py [hunk 3]
```diff
@@ -108,6 +118,15 @@ def confidence(self) -> bool:
         """Returns true if the model is able to return a confidence score along with its predictions"""
         return self._model_config.get(self.COMPUTE_CONFIDENCE_KEY, False)
 
+    # Embedding config
+    def embedding_provider(self) -> str:
```
Comment (Abhinav-Naikawadi) [inline src/autolabel/configs/config.py:122]: We use the LLM provider when an embedding model provider is not specified

**Comment by nihit** [comment]
Addresses #370

**Review by nihit — CHANGES_REQUESTED** [review]
@Abhinav-Naikawadi  
1. please add example configs in the PR description here for using HuggingFace and VertexAI embeddings for one of the benchmark tasks.
2. Add tests for verifying the embedding config is correctly used in the few shot initialization class
3. Make relevant changes to config schema for the new embedding key - https://github.com/refuel-ai/autolabel/blob/main/src/autolabel/configs/schema.py

**Comment by Abhinav-Naikawadi** [comment]
Example config for huggingface embeddings:

{
    "task_name": "BankingComplaintsClassification",
    "task_type": "classification",
    "dataset": {
        "label_column": "label",
        "delimiter": ","
    },
    "model": {
        "provider": "huggingface_pipeline",
        "name": "google/flan-t5-small"
    },
    "embedding": {
        "provider": "huggingface_pipeline",
        "model": "sentence-transformers/all-mpnet-base-v2"
    },
    ...

**Comment by Abhinav-Naikawadi** [comment]
Example config with google (vertexai) embeddings:
{
    "task_name": "BankingComplaintsClassification",
    "task_type": "classification",
    "dataset": {
        "label_column": "label",
        "delimiter": ","
    },
    "model": {
        "provider": "google",
        "name": "gpt-3.5-turbo"
    },
    "embedding": {
        "provider": "google"
    },
    ...

**Review by nihit — APPROVED** [review]

**Comment by nihit** [comment]
can merge once tests passing validated

**Pull request merged**